### ISSUE TITLE
switching the optimizer from adam to LBFGSB

### ISSUE THREAD

**Issue by SeoheeJang1** [sciann/sciann#88]
Dear all,

Hi

when I use the optimizer, many literatures adopt switching the adam optimizer to L-BFGS-B,

I can do separately, use those optimizers but how can I switch during training session

I made callback function with on_epoch_begin or on_epoch_end function in the keras callback function

since Scimodel does not have optimizer attribute, I can't switch the optimizer properly

can you give me some advices for this issue??

**Comment by adxtop1** [comment]
Hey can help me in switching the optimizer from adam to L-BFGS-B separately. I am unable to do that too. Any reference material would be highly appreciated

**Comment by ehsanhaghighat** [comment]
use ’scipy-XXX’ with XXX a name from scipy minimize optimizers. e.g.:
>> ’scipy-L-BFGS-B’

**Comment by adxtop1** [comment]
Could you share the example plzzz.

**Comment by adxtop1** [comment]
Hey gotcha. Thanksss

**Comment by ehsanhaghighat** [comment]
We have lots of examples in our repo. Also, feel free to create a simple colab example then I can point you different settings.

**Comment by SeoheeJang1** [comment]
so is there any way to convert adam to LBFGSB in sequentially??

**Comment by adxtop1** [comment]
I think you can save the weights after training with Adam. Now change the optimizer to LBFGSB and reinitialize the weight of the model with earlier weights and then train.

**Comment by SeoheeJang1** [comment]
thank you for replying, 
so you mean that save the weight after training with adam and make the new scimodel which has optimizer LBFGSB and load the weight from adam and train again right?

**Comment by adxtop1** [comment]
Yess, though it's a lengthy process but it should result in the desired behaviour. I tried it that way for my models.

**Comment by SeoheeJang1** [comment]
Okay I though there is other way to do it during training with callback function. 
But Yeah thank you for letting me know

**Comment by adxtop1** [comment]
Hey on a completely different note. Are you aware of any method for inverse
problem where I can predict larger values? Like for smaller values i am
able to get correct values of Lambda predicted by my model but for large
values like 400, 500 my lambda become 0. I know it is out of context but if
you know anything just let me know.

**Comment by SeoheeJang1** [comment]
I think you can set the min max value to confine the sn.parameter so that it has hard constraints to get the value in that range 
or normalisation could be other problem as well

**Comment by adxtop1** [comment]
Thanks, I'll check it out.
